File: upf/python/clients/java/JClient/src/main/java/eu/aiplan4eu/jclient/Problem.java
package eu.aiplan4eu.jclient;

import java.util.Set;

public class Problem {

    private Set<Action> actions;
    private Set<String> init;
    private Set<String> goal;

    public Problem(Set<Action> actions, Set<String> init, Set<String> goal) {
        this.actions = actions;
        this.init = init;
        this.goal = goal;
    }

    public Set<Action> getActions() {
        return actions;
    }

    public Set<String> getInit() {
        return init;
    }

    public Set<String> getGoal() {
        return goal;
    }

}

File: upf/cpp/problem.cxx
#include "problem.hxx"

namespace upf
{

  Problem::Problem(const std::vector<Action>& actions,
                   const std::set<std::string>& init,
                   const std::set<std::string>& goal)
  {
    actions_ = actions;
    init_ = init;
    goal_ = goal;
  }

  const std::vector<Action>& Problem::actions() const
  {
    return actions_;
  }

  const std::set<std::string>& Problem::init() const
  {
    return init_;
  }

  const std::set<std::string>& Problem::goal() const
  {
    return goal_;
  }

}
